src/pudl/etl/epacems_assets.py

```diff
@@ -13,7 +14,25 @@
 logger = pudl.logging_helpers.get_logger(__name__)
 
 
-@asset(
+YearPartitions = namedtuple("YearPartitions", ["year", "states"])
+
+
+@op(
+    out=DynamicOut(),
```

Comment:
If/when Dagster gets partitioned assets working with unpartitioned inputs, do we think there's a clear path to making this asset behave like all the others?

What would our IO Managers need to look like to make that work? Why was the existing Parquet IO Manager so inefficient for the purpose of parallelizing CEMS?

Reply:
I think using the same patterns / abstractions across the board will make the project more legible and maintainable, so creating an appropriate IO Manager seems like a good idea, but for now I think we should probably get this merged in and focus on getting all the outputs into the database.  We can revisit managing Parquet outputs uniformly when we encounter another Parquet dataset like FERC EQR.

I suspect that the dask dataframe filtering could also be made much more efficient if we put the right metadata in the Parquet files and are only filtering based on our row-groups (year and state in the case of EPA CEMS).